frontend/www/js/omegaup/arena/contest_practice.ts

```diff
@@ -109,14 +120,36 @@ OmegaUp.on('ready', () => {
       });
   }
 
-  // The hash is of the form `#problems/${alias}`.
-  const problemMatch = /#problems\/([^/]+)/.exec(window.location.hash);
-  const problemAlias = problemMatch?.[1] ?? null;
-  if (problemAlias) {
-    // This needs to be set here and not at the top because it depends
-    // on the `navigate-to-problem` callback being invoked, and that is
-    // not the case if this is set a priori.
-    contestPractice.problem = { alias: problemAlias, runs: [] };
+  // The hash is of the forms:
+  // - `#problems/${alias}`
+  // - `#problems/${alias}/new-run`
+  // - `#problems/${alias}/show-run:xyz`
+  // - `#clarifications/${alias}/new`
+  // and all the matching forms in the following regex
+  const match = /#(?<tab>\w+)\/(?<alias>[^/]+)(?:\/(?<popup>[^/]+))?/g.exec(
+    window.location.hash,
+  );
+  switch (match?.groups?.tab) {
+    case 'problems':
+      // This needs to be set here and not at the top because it depends
+      // on the `navigate-to-problem` callback being invoked, and that is
+      // not the case if this is set a priori.
+      contestPractice.problem = { alias: match.groups.problem, runs: [] };
+      if (match.groups.popup === 'new-run') {
+        contestPractice.popupDisplayed = PopupDisplayed.RunSubmit;
+      }
+      if (match.groups.popup.startsWith('show-run')) {
```

Comment:
```suggestion
      } else if (match.groups.popup.startsWith('show-run')) {
```

son mutuamente exclusivos, así que mejor hay que denotarlo.

Reply:
Listo. Se me pasó en la revisión anterior, perdón.